romancal/tweakreg/tests/test_tweakreg.py

```diff
@@ -301,15 +318,24 @@ def add_tweakreg_catalog_attribute(
     if catalog_data is None:
         catalog_data = get_catalog_data(input_dm)
 
-    create_base_image_source_catalog(
+    source_catalog = create_base_image_source_catalog(
         tmp_path,
         tweakreg_catalog_filename,
         catalog_data=catalog_data,
         catalog_format=catalog_format,
+        save_catalogs=save_catalogs,
     )
-    input_dm.meta["tweakreg_catalog"] = os.path.join(
-        tmp_path, tweakreg_catalog_filename
-    )
+    if save_catalogs:
+        input_dm.meta["source_detection"] = {
+            "tweakreg_catalog": source_catalog,
+            "tweakreg_catalog_name": "",
+        }
+    else:
+        input_dm.meta["source_detection"] = {
+            "tweakreg_catalog": None,
+            "tweakreg_catalog_name": os.path.join(tmp_path, tweakreg_catalog_filename),
```

Comment:
If `save_catalogs=True`, then that means we also want the source catalog saved on disk, in which case `meta.source_detection.tweakreg_catalog_name` will be updated to point to where the source catalog is located (so that `TweakRegStep` can read the file in).